dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/icon/IconTest.java

```diff
@@ -87,12 +88,30 @@ protected void initTest() throws IOException {
   void shouldGetAllIconsWhenRequested() {
     Map<String, DefaultIcon> defaultIconMap = getAllDefaultIcons();
 
+    IconCriteria iconCriteria = new IconCriteria();
+    iconCriteria.setType(IconType.ALL);
+    iconCriteria.setKeywords(List.of());
+
+    assertEquals(
+        defaultIconMap.size() + 1,
+        iconService.getIcons(iconCriteria).size(),
+        String.format(
+            "Expected to find %d icons, but found %d instead",
+            defaultIconMap.size() + 1, iconService.getIcons(iconCriteria).size()));
+  }
+
+  @Test
+  void shouldGetAllIcons() {
```

Comment:
What is the difference between this and the above test? Should this show that `shouldGetAllIconsByDefault`? if so I would rename the test. I also wonder if we need an integration test for this as this only tests IconCriteria.type defaults to ALL

Reply:
fixed.